app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,6 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, only: %i[new create]
+
   def show
     @post = Post.find(params[:id])
     @index = params[:index]
```

Comment:
- [ ] Great job adding cancancan for authorization. You've done great work here. :handshake: . However, there's little stuff to add here for the properly implementation of cancancan in your app.
In your post controller, use the load_and_authorize_resource method to load and authorize resources. This should be placed at the top of your post controller class.

```
  class PostsController < ApplicationController
     load_and_authorize_resource
     -
     -
     -
  
 end

```

By using load_and_authorize_resource, CanCanCan will load the appropriate resource based on the controller and action name, and it will also authorize the action based on the defined abilities in your ability.rb file. If the user doesn't have the necessary permissions, it will raise a CanCan::AccessDenied exception that you can handle appropriately (e.g., by redirecting to an error page).

Reply:
- Well resolved 👍🏽